ext/ManoptJuMPExt.jl

```diff
@@ -0,0 +1,440 @@
+module ManoptJuMPExt
+
+using Manopt
+using LinearAlgebra
+if isdefined(Base, :get_extension)
+    using JuMP: JuMP
+else
+    # imports need to be relative for Requires.jl-based workflows:
+    # https://github.com/JuliaArrays/ArrayInterface.jl/pull/387
+    using ..JuMP: JuMP
+end
+const MOI = JuMP.MOI
+using ManifoldsBase
+using ManifoldDiff
+
+function __init__()
+    # So that the user can use the convenient `Manopt.Optimizer`
+    if isdefined(Base, :setglobal!)
+        setglobal!(Manopt, :JuMP_Optimizer, Optimizer)
+        setglobal!(Manopt, :JuMP_VectorizedManifold, VectorizedManifold)
```

Comment:
Is it necessary that `VectorizedManifold` and `JuMP_ArrayShape` are visible outside the extension?

to me, the `setglobal!` alone already seems to be a hack, now with the eval alternative even a bit more. If we would have a less-tricky variant that would be great (but I do not completely see why they have to be that visible, at least the last 2))

Reply:
It's only so that they can be included as docstring in Documenter.

Reply:
And you seemed to suggest that otherwise it is not possible to access them ?

Reply:
This is a hack but we need this to get docstrings in documentation, at least until Documenter.jl figures out a better way.

Reply:
Ok, if that is only for documenter, then that Is fine with me as well.